buffer.c

```diff
@@ -140,6 +141,35 @@ buffer_resize(buffer_t *self, size_t n) {
 }
 
 /*
+ * Append a printf-style formatted string to the buffer.
+ */
+int buffer_appendf(buffer_t *self, const char *format, ...) {
+  va_list ap;
+  va_start(ap, format);
+  const int initial_len = buffer_length(self);
+
+  // First, we compute how many bytes are needed for the formatted string
+  // and allocate that much more space in the buffer.
+  va_list tmpa;
+  va_copy(tmpa, ap);
+  const int space_required = vsnprintf(NULL, 0, format, tmpa);
+  va_end(tmpa);
+  const int resized = buffer_resize(self, initial_len + space_required);
+  if (resized == -1) {
+    va_end(ap);
+    return -1;
+  }
+
+  // Next format the string into the space that we have made room for.
+  char* dst = self->data + initial_len;
+  const int bytes_formatted = vsnprintf(dst, 1+space_required, format, ap);
+
+  printf("Result '%s'\n", self->data);
+  va_end(ap);
+  return (bytes_formatted < 0) ? -1 : 0;
```

Comment:
well the parentheses are unnecessary, but they dont hurt anything either.. i generally format ternary statements like:

```c
return a > b
  ? c
  : d
```

but i'm not sure it makes anything "cleaner".

---

i think the `return` is fine